lib/elixir/lib/uri.ex

```diff
@@ -71,44 +71,67 @@ defmodule URI do
   end
 
   @doc """
-  Encodes an enumerable into a query string.
+  Encodes `enumerable` into a query string using `encoding`.
 
   Takes an enumerable that enumerates as a list of two-element
   tuples (for instance, a map or a keyword list) and returns a string
-  in the form of `key1=value1&key2=value2...` where keys and
-  values are URL encoded as per `encode_www_form/1`.
+  in the form of `key1=value1&key2=value2...`.
 
   Keys and values can be any term that implements the `String.Chars`
   protocol with the exception of lists, which are explicitly forbidden.
 
+  You can specify one of the following `encoding` strategies:
+
+    * `:www_form` - (default) keys and values are URL encoded as per
+      `encode_www_form/1`. This is the format typically used by browsers
+      on query strings and form data. It encodes " " as "+".
+
+    * `:rfc_3986` - the same as `:www_form` except it encodes " " as
+      "%20" according [RFC 3986](https://tools.ietf.org/html/rfc3986).
+      This is the best option if you are encoding a non-http related
```

Comment:
Considearion: "non-http scheme" might give the impression that "+" space encoding is OK for all http(s) situations. I ran into a "+ interpreted as literal plus" situation while connecting to an HTTP(S) web API. If I understand the specs correctly, even for HTTP URI's, this is technically correct behaviour.

Maybe something like "non-browser situation" would be better?

Reply:
Sounds good, please change! :)

Reply:
Changed in https://github.com/elixir-lang/elixir/pull/10680/commits/941c76bb2d3b8ca59e0f88ce1a6ee9a9061943e2